components/post_view/post_body/post_body.tsx

```diff
@@ -146,7 +158,7 @@ export default class PostBody extends React.PureComponent {
         }
 
         let failedOptions;
-        if (this.props.post.failed) {
+        if (this.props.post && this.props.post.failed) {
```

Comment:
wouldn't this be more idiomatic like 
```suggestion
        if (this.props.post!.failed) {
```